src/overlays/effects/ovl_Effect_Ss_Solder_Srch_Ball/z_eff_ss_solder_srch_ball.c

```diff
@@ -1,27 +1,89 @@
 /*
  * File: z_eff_ss_solder_srch_ball.c
  * Overlay: ovl_Effect_Ss_Solder_Srch_Ball
- * Description:
+ * Description: Vision sphere
  */
 
 #include "z_eff_ss_solder_srch_ball.h"
+#include "objects/gameplay_keep/gameplay_keep.h"
+
+#define rFlags regs[0]
 
 #define PARAMS ((EffectSsSolderSrchBallInitParams*)initParamsx)
 
-s32 EffectSsSolderSrchBall_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
+u32 EffectSsSolderSrchBall_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx);
 void EffectSsSolderSrchBall_Update(PlayState* play, u32 index, EffectSs* this);
 void EffectSsSolderSrchBall_Draw(PlayState* play, u32 index, EffectSs* this);
 
-#if 0
 const EffectSsInit Effect_Ss_Solder_Srch_Ball_InitVars = {
     EFFECT_SS_SOLDER_SRCH_BALL,
     EffectSsSolderSrchBall_Init,
 };
 
-#endif
+u32 EffectSsSolderSrchBall_Init(PlayState* play, u32 index, EffectSs* this, void* initParamsx) {
+    EffectSsSolderSrchBallInitParams* initParams = PARAMS;
+
+    this->pos = initParams->pos;
+    this->velocity = initParams->velocity;
+    this->accel = initParams->accel;
+    this->update = EffectSsSolderSrchBall_Update;
+    if (!(initParams->flags & SOLDERSRCHBALL_INVISIBLE)) {
+        this->draw = EffectSsSolderSrchBall_Draw;
+    }
+    this->life = 10;
+    this->rgScale = initParams->scale;
+    this->rFlags = initParams->flags;
+    this->actor = (Actor*)initParams->playerDetected; // actor field was incorrectly used as a pointer to something else
```

Comment:
Does anything else expect to read this struct's actor field as an actor? If so, it's a bug.